### DISCUSSION TITLE
Highly generic cosmetic filter

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBOL-home#55] [Q&A] [answered]
https://github.com/uBlockOrigin/uBOL-issues/issues/54#issuecomment-1622193653:

> Whether highly generic cosmetic filters should be supported is something I haven't yet decided.

Is there some thing filterlist maintainers could do to know if a cosmetic filter is highly generic?

**Comment by gorhill** [comment, marked as answer]
Highly generic cosmetic filters are those with no id `#` and no class `.` anywhere in their selector:

    ##[href^="https://sbobet.host"]
    ##a[href^="https://sbobet.host"]

If there appears at least one id or class, the cosmetic filter will be lowly generic and injected through the DOM surveyor.

Not highly generic

    ###a [href^="https://sbobet.host"]
    ###a[href^="https://sbobet.host"]
    ##.a [href^="https://sbobet.host"]
    ##.a[href^="https://sbobet.host"]
    ##[href^="https://sbobet.host"] #b
    ##[href^="https://sbobet.host"].b
    ...

**Reply by stephenhawk8054** [reply]
How about the filters that just have other attributes (but not contain attributes' values), like `##[data-ad-cls]`?

**Reply by gorhill** [reply]
Highly generic since they do not have any id or class.

**Reply by stephenhawk8054** [reply]
I see, understood.
